sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/SetFns.java

```diff
@@ -0,0 +1,261 @@
+        (numberOfElementsinLeft, numberOfElementsinRight) -> (numberOfElementsinLeft > 0 && numberOfElementsinRight > 0) ? Math.min(numberOfElementsinLeft, numberOfElementsinRight) : 0L;
+    return new SetImpl<>(rightCollection, intersectFn);
+  }
+
+  /**
+   * Returns a new {@code SetFns.SetImpl<T>} transform that compute the difference (except) with
+   * provided {@code PCollection<T>}.
+   *
+   * <p>The argument should not be modified after this is called.
+   *
+   * <p>The elements of the output {@link PCollection<T>} will all distinct elements that present in
+   * pipeline is constructed {@link PCollection<T>} but not present in provided {@link
+   * PCollection<T>}.
+   *
+   * <pre>{@code
+   * Pipeline p = ...;
+   *
+   * PCollection<String> left = p.apply(Create.of("1", "2", "3", "4", "5"));
+   * PCollection<String> right = p.apply(Create.of("1", "3", "4", "6"));
+   *
+   * PCollection<String> results =
+   *     left.apply(SetFns.except(right));
+   * }</pre>
+   */
+  public static <T> SetImpl<T> except(PCollection<T> rightCollection) {
+    checkNotNull(rightCollection, "rightCollection argument is null");
+    SerializableBiFunction<Long, Long, Long> exceptFn =
+        (numberOfElementsinLeft, numberOfElementsinRight) -> numberOfElementsinLeft > 0 && numberOfElementsinRight == 0 ? 1L : 0L;
+    return new SetImpl<>(rightCollection, exceptFn);
+  }
+
+  /**
+   * Returns a new {@code SetFns.SetImpl<T>} transform that compute the difference all (exceptAll)
+   * with provided {@code PCollection<T>}.
+   *
+   * <p>The argument should not be modified after this is called.
+   *
+   * <p>The elements of the output {@link PCollection<T>} which will follow EXCEPT_ALL Semantics as
+   * follows: Given there are m elements on pipeline which is constructed {@link PCollection<T>}
+   * (left) and n elements on in provided {@link PCollection<T>} (right): - it will output m
+   * elements of left for all elements which are present in left but not in right. - it will output
+   * MAX(m - n, 0) elements of left for all elements which are present in both left and right.
+   *
+   * <pre>{@code
+   * Pipeline p = ...;
+   *
+   * PCollection<String> left = p.apply(Create.of("1", "2", "3", "4", "5"));
+   * PCollection<String> right = p.apply(Create.of("1", "3", "4", "6"));
+   *
+   * PCollection<String> results =
+   *     left.apply(SetFns.exceptAll(right));
+   * }</pre>
+   */
+  public static <T> SetImpl<T> exceptAll(PCollection<T> rightCollection) {
+    checkNotNull(rightCollection, "rightCollection argument is null");
+    SerializableBiFunction<Long, Long, Long> exceptFn =
+        (numberOfElementsinLeft, numberOfElementsinRight) -> {
+          if (numberOfElementsinLeft > 0 && numberOfElementsinRight == 0) {
+            return numberOfElementsinLeft;
+          } else if (numberOfElementsinLeft > 0 && numberOfElementsinRight > 0) {
+            return Math.max(numberOfElementsinLeft - numberOfElementsinRight, 0L);
```

Comment:
Isn't this expression always sufficient?

Reply:
Thanks very valuable. I have fixed this.